lib/google_service_client.rb

```diff
@@ -0,0 +1,91 @@
+##
+# module of generic GCP-centric methods, such as handling authorization/tokens, or GCS drivers
+##
+module GoogleServiceClient
+  ##
+  # OAuth token methods
+  ##
+
+  # refresh access_token when expired and stores back in instance of parent class
+  #
+  # * *return*
+  #   - +DateTime+ timestamp of new access token expiration
+  def refresh_access_token!
```

Comment:
Same, I'd make this a class method.  (and you can always get the callee by introspection if you want it for logging purposes.)

Reply:
I don't actually want this to be a class method as the classes that this gets included in are all singletons. 
 `FireCloudClient`, `PapiClient`, and `DataRepoClient` each are instantiated as class variables for `ApplicationController` - we do this specifically so that everyone uses the same instance, and share the same access token.  Therefore, when I call `ApplicationController.firecloud_client.refresh_access_token!`, it updates the access token _in place_ for that shared instance of `FireCloudClient`.  If we don't do this, eventually we'd run afoul of the Google quota on generating access tokens, and everything would break until the lock expires.

Reply:
Exactly, that's why this should be a class method (self.refresh_access_token). If this GoogleServiceClient is included in 3 other classes, having this be an instance method implies that each of them have their own token, and we have 3 separate Google tokens.  Which isn't what we want, right?

Reply:
Essentially, I was expecting this class to be the 'Google access token manager', and other classes call it when they need a token.  And so I wouldn't necessarily include in other services at all -- it would be a singleton class of its own.